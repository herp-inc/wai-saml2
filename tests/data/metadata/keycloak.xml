<?xml version="1.0" encoding="UTF-8"?>
<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" entityID="https://sso.ja-sore.de/auth/realms/HERP">
   <md:IDPSSODescriptor WantAuthnRequestsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
      <md:KeyDescriptor use="signing">
         <ds:KeyInfo>
            <ds:KeyName>AViu9Cf7_rjRHwqkm_MgvMcjWHuRPSP0mMNNEjHslaY</ds:KeyName>
            <ds:X509Data>
               <ds:X509Certificate>MIIClzCCAX8CBgF3TK79BzANBgkqhkiG9w0BAQsFADAPMQ0wCwYDVQQDDARIRVJQMB4XDTIxMDEyOTA1NDYxMloXDTMxMDEyOTA1NDc1MlowDzENMAsGA1UEAwwESEVSUDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBANRBHQMqF2+g7CrESyzrv+HvwjvIdhdZWWc6z4GiD3zN1TE+cbh88JMUrVTWd7EJLq+9uAazCyZfzX5BdauDqsp+LP9z2xiJ6Zlao1z5ycfG4dwvEIlOv4rX1zSdtg35Cz+QdaUfIkCEX3dTHI7dFcwou+Nel0BGCnp17w8gVPOBGYzvOtBgWcNV9+9Zmn7oSRoG5R4uRzhmPv12pyKRO/0hZr6gTkQ11CsnyfW890bDM05I5m8gDT0fAO2PznbQIQKClqUoyNbJZbBeDyV7QaEkkC8Nckvo6y+bae8oEF9pxQiXBV7dHtY8VLWq5SN2IeX1VK59hMg2NdHUxjMZIiECAwEAATANBgkqhkiG9w0BAQsFAAOCAQEAzGn04cz8oaZoRUCo6OKhcwfOSrjojfqbUxW8ZAY6UPUV8IfAbaxMlc4j1NVq3awi+yCJWLDaZyOX4mQyD+NoHB5D1/qQi/wLBkg9TJt2wp6MjkOPrwkN8lvhabq8Us2NiBnhHzLjWawltie9r1W0pugRG4qSZvJqosJJh9Wr0m8MnaP28P7fp4J/j3UAivlGZvXXh1kcl/lRWhoYqmR3r2j98OWy/WdYfx/oPNDWMdJDdR9I/qtlOVHsiEDP8Z478/z9IYweOYYqg7Ou77Ys/5G6wRxRCwaUjMDPdFfyvfUBZ9EFpYUV1G/zXuGQwfuh27GWz8gytdd1vDxBU06h3w==</ds:X509Certificate>
            </ds:X509Data>
         </ds:KeyInfo>
      </md:KeyDescriptor>
      <md:ArtifactResolutionService Binding="urn:oasis:names:tc:SAML:2.0:bindings:SOAP" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml/resolve" index="0" />
      <md:SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</md:NameIDFormat>
      <md:NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:transient</md:NameIDFormat>
      <md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified</md:NameIDFormat>
      <md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</md:NameIDFormat>
      <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:SOAP" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
      <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" Location="https://sso.ja-sore.de/auth/realms/HERP/protocol/saml" />
   </md:IDPSSODescriptor>
</md:EntityDescriptor>
