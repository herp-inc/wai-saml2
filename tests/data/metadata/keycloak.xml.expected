IDPSSODescriptor
  { entityID = "https://sso.ja-sore.de/auth/realms/HERP"
  , x509Certificate =
      SignedExact
        { getSigned =
            Signed
              { signedObject =
                  Certificate
                    { certVersion = 0
                    , certSerial = 1611899272455
                    , certSignatureAlg = SignatureALG HashSHA256 PubKeyALG_RSA
                    , certIssuerDN =
                        DistinguishedName
                          { getDistinguishedElements =
                              [ ( [ 2 , 5 , 4 , 3 ]
                                , ASN1CharacterString
                                    { characterEncoding = UTF8
                                    , getCharacterStringRawData = "HERP"
                                    }
                                )
                              ]
                          }
                    , certValidity =
                        ( DateTime
                            { dtDate =
                                Date { dateYear = 2021 , dateMonth = January , dateDay = 29 }
                            , dtTime =
                                TimeOfDay
                                  { todHour = 5 h , todMin = 46 m , todSec = 12 s , todNSec = 0 ns }
                            }
                        , DateTime
                            { dtDate =
                                Date { dateYear = 2031 , dateMonth = January , dateDay = 29 }
                            , dtTime =
                                TimeOfDay
                                  { todHour = 5 h , todMin = 47 m , todSec = 52 s , todNSec = 0 ns }
                            }
                        )
                    , certSubjectDN =
                        DistinguishedName
                          { getDistinguishedElements =
                              [ ( [ 2 , 5 , 4 , 3 ]
                                , ASN1CharacterString
                                    { characterEncoding = UTF8
                                    , getCharacterStringRawData = "HERP"
                                    }
                                )
                              ]
                          }
                    , certPubKey =
                        PubKeyRSA
                          PublicKey
                            { public_size = 256
                            , public_n =
                                26794629232301390559329251866062548351573676959105948670047115599484903764153173211872085488745203643849524181464124430648664560428395941550339690556064181777927415101879096439027796505139145202723504898654935482393722978364774771646139306434335595822802292206982564879345724045673514452796802353368762193429772204243864488315862304970769841689431483551389268024017379114959044330977741690398490550580701831871711628065022359440646669330356587937408829251700791982308190330679272353052846282673680829331107723412188694305482396334605009242290197499924909025290277992294341810791131897406148044363907966374411188642337
                            , public_e = 65537
                            }
                    , certExtensions = Extensions Nothing
                    }
              , signedAlg = SignatureALG HashSHA256 PubKeyALG_RSA
              , signedSignature =
                  "\204i\244\225\204\252\161\166hE@\168\232\226\161s\a\206J\184\232\141\250\155S\NAK\188d\ACK:P\245\NAK\240\135\192m\172L\149\206#\212\213j\221\172\"\251 \137X\176\218g#\151\226d2\SI\227h\FS\RSC\215\250\144\139\252\v\ACKH=L\155v\194\158\140\142C\143\175\t\r\242[\225i\186\188R\205\141\136\EM\225\US2\227Y\172%\182'\189\175U\180\166\232\DC1\ESC\138\146f\242j\162\194I\135\213\171\210o\f\157\163\246\240\254\223\167\130\DEL\143u\NUL\138\249Ff\245\215\135Y\FS\151\249QZ\SUB\CAN\170dw\175h\253\240\229\178\253gX\DEL\US\232<\208\214\&1\210Cu\USH\254\171e9Q\236\136@\207\241\158;\243\252\253!\140\RS9\134*\131\179\174\239\182,\255\145\186\193\FSQ\v\ACK\148\140\192\207tW\242\189\245\SOHg\209\ENQ\165\133\NAK\212o\243^\225\144\193\251\161\219\177\150\207\200\&2\181\215u\188<ASN\161\223"
              }
        , exactObjectRaw =
            "0\130\SOH\DEL\STX\ACK\SOHwL\174\253\a0\r\ACK\t*\134H\134\247\r\SOH\SOH\v\ENQ\NUL0\SI1\r0\v\ACK\ETXU\EOT\ETX\f\EOTHERP0\RS\ETB\r210129054612Z\ETB\r310129054752Z0\SI1\r0\v\ACK\ETXU\EOT\ETX\f\EOTHERP0\130\SOH\"0\r\ACK\t*\134H\134\247\r\SOH\SOH\SOH\ENQ\NUL\ETX\130\SOH\SI\NUL0\130\SOH\n\STX\130\SOH\SOH\NUL\212A\GS\ETX*\ETBo\160\236*\196K,\235\191\225\239\194;\200v\ETBYYg:\207\129\162\SI|\205\213\&1>q\184|\240\147\DC4\173T\214w\177\t.\175\189\184\ACK\179\v&_\205~Au\171\131\170\202~,\255s\219\CAN\137\233\153Z\163\\\249\201\199\198\225\220/\DLE\137N\191\138\215\215\&4\157\182\r\249\v?\144u\165\US\"@\132_wS\FS\142\221\NAK\204(\187\227^\151@F\nzu\239\SI T\243\129\EM\140\239:\208`Y\195U\247\239Y\154~\232I\SUB\ACK\229\RS.G8f>\253v\167\"\145;\253!f\190\160ND5\212+'\201\245\188\247F\195\&3NH\230o \r=\US\NUL\237\143\206v\208!\STX\130\150\165(\200\214\201e\176^\SI%{A\161$\144/\rrK\232\235/\155i\239(\DLE_i\197\b\151\ENQ^\221\RS\214<T\181\170\229#v!\229\245T\174}\132\200\&65\209\212\198\&3\EM\"!\STX\ETX\SOH\NUL\SOH"
        , encodeSignedObject =
            "0\130\STX\151\&0\130\SOH\DEL\STX\ACK\SOHwL\174\253\a0\r\ACK\t*\134H\134\247\r\SOH\SOH\v\ENQ\NUL0\SI1\r0\v\ACK\ETXU\EOT\ETX\f\EOTHERP0\RS\ETB\r210129054612Z\ETB\r310129054752Z0\SI1\r0\v\ACK\ETXU\EOT\ETX\f\EOTHERP0\130\SOH\"0\r\ACK\t*\134H\134\247\r\SOH\SOH\SOH\ENQ\NUL\ETX\130\SOH\SI\NUL0\130\SOH\n\STX\130\SOH\SOH\NUL\212A\GS\ETX*\ETBo\160\236*\196K,\235\191\225\239\194;\200v\ETBYYg:\207\129\162\SI|\205\213\&1>q\184|\240\147\DC4\173T\214w\177\t.\175\189\184\ACK\179\v&_\205~Au\171\131\170\202~,\255s\219\CAN\137\233\153Z\163\\\249\201\199\198\225\220/\DLE\137N\191\138\215\215\&4\157\182\r\249\v?\144u\165\US\"@\132_wS\FS\142\221\NAK\204(\187\227^\151@F\nzu\239\SI T\243\129\EM\140\239:\208`Y\195U\247\239Y\154~\232I\SUB\ACK\229\RS.G8f>\253v\167\"\145;\253!f\190\160ND5\212+'\201\245\188\247F\195\&3NH\230o \r=\US\NUL\237\143\206v\208!\STX\130\150\165(\200\214\201e\176^\SI%{A\161$\144/\rrK\232\235/\155i\239(\DLE_i\197\b\151\ENQ^\221\RS\214<T\181\170\229#v!\229\245T\174}\132\200\&65\209\212\198\&3\EM\"!\STX\ETX\SOH\NUL\SOH0\r\ACK\t*\134H\134\247\r\SOH\SOH\v\ENQ\NUL\ETX\130\SOH\SOH\NUL\204i\244\225\204\252\161\166hE@\168\232\226\161s\a\206J\184\232\141\250\155S\NAK\188d\ACK:P\245\NAK\240\135\192m\172L\149\206#\212\213j\221\172\"\251 \137X\176\218g#\151\226d2\SI\227h\FS\RSC\215\250\144\139\252\v\ACKH=L\155v\194\158\140\142C\143\175\t\r\242[\225i\186\188R\205\141\136\EM\225\US2\227Y\172%\182'\189\175U\180\166\232\DC1\ESC\138\146f\242j\162\194I\135\213\171\210o\f\157\163\246\240\254\223\167\130\DEL\143u\NUL\138\249Ff\245\215\135Y\FS\151\249QZ\SUB\CAN\170dw\175h\253\240\229\178\253gX\DEL\US\232<\208\214\&1\210Cu\USH\254\171e9Q\236\136@\207\241\158;\243\252\253!\140\RS9\134*\131\179\174\239\182,\255\145\186\193\FSQ\v\ACK\148\140\192\207tW\242\189\245\SOHg\209\ENQ\165\133\NAK\212o\243^\225\144\193\251\161\219\177\150\207\200\&2\181\215u\188<ASN\161\223"
        }
  , nameIDFormats =
      [ "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent"
      , "urn:oasis:names:tc:SAML:2.0:nameid-format:transient"
      , "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
      , "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
      ]
  , singleSignOnServices =
      [ ( HTTPPost
        , "https://sso.ja-sore.de/auth/realms/HERP/protocol/saml"
        )
      , ( HTTPRedirect
        , "https://sso.ja-sore.de/auth/realms/HERP/protocol/saml"
        )
      , ( SOAP
        , "https://sso.ja-sore.de/auth/realms/HERP/protocol/saml"
        )
      , ( HTTPArtifact
        , "https://sso.ja-sore.de/auth/realms/HERP/protocol/saml"
        )
      ]
  }